readthedocs/projects/tasks.py

```diff
@@ -1282,13 +1282,8 @@ def fileify(version_pk, commit, build):
     except Exception:
         log.exception('Failed during ImportedFile creation')
 
-    try:
-        _update_intersphinx_data(version, path, commit, build)
```

Comment:
I think we should keep this as 3 functions called from here, instead of one. That will make it more robust to issues of exceptions. So we should have:
* _create_imported_files
* _create_intersphinx_data
* _sync_imported_files

And refactor the code to match